plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/HttpThriftMetastoreClientFactory.java

```diff
@@ -0,0 +1,154 @@
+
+    @Inject
+    public HttpThriftMetastoreClientFactory(
+            ThriftHttpMetastoreConfig httpMetastoreConfig,
+            NodeManager nodeManager)
+    {
+        this(
+                httpMetastoreConfig.getReadTimeout(),
+                new ThriftHttpContext(httpMetastoreConfig.getAuthenticationMode(), httpMetastoreConfig.getHttpBearerToken(), httpMetastoreConfig.getAdditionalHeaders()),
+                nodeManager.getCurrentNode().getHost());
+    }
+
+    @Override
+    public ThriftMetastoreClient create(URI uri, Optional<String> delegationToken)
+            throws TTransportException
+    {
+        return new ThriftHiveMetastoreClient(
+                () -> createHttpTransport(uri, thriftHttpContext),
+                hostname);
+    }
+
+    private TTransport createHttpTransport(URI uri, ThriftHttpContext httpThriftContext)
+            throws TTransportException
+    {
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        if ("https".equals(uri.getScheme().toLowerCase(ENGLISH))) {
+            checkArgument(httpThriftContext.token.isPresent(), "'hive.metastore.http.client.bearer-token' must be set while using https metastore URIs in 'hive.metastore.uri'");
+            checkArgument(httpThriftContext.authenticationMode.isPresent(), "'hive.metastore.http.client.authentication.type' must be set while using http/https metastore URIs in 'hive.metastore.uri'");
+            SSLConnectionSocketFactory socketFactory;
+            try {
+                socketFactory = new SSLConnectionSocketFactory(SSLContext.getDefault(), new DefaultHostnameVerifier(null));
+            }
+            catch (NoSuchAlgorithmException e) {
+                throw new TTransportException(e);
+            }
+            Registry<ConnectionSocketFactory> registry = RegistryBuilder.<ConnectionSocketFactory>create()
+                    .register("https", socketFactory)
+                    .build();
+            httpClientBuilder.setConnectionManager(new BasicHttpClientConnectionManager(registry));
+            httpClientBuilder.addRequestInterceptorFirst((httpRequest, entityDetails, httpContext) -> {
+                httpRequest.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + httpThriftContext.token.get());
+            });
+        }
+        else {
+            checkArgument(httpThriftContext.token.isEmpty(), "'hive.metastore.http.client.bearer-token' must not be set while using http metastore URIs in 'hive.metastore.uri'");
+        }
+        httpClientBuilder.addRequestInterceptorFirst((httpRequest, entityDetails, httpContext) -> {
+            httpThriftContext.additionalHeaders().forEach(httpRequest::addHeader);
+        });
+        httpClientBuilder.setDefaultRequestConfig(RequestConfig.custom().setResponseTimeout(readTimeoutMillis, TimeUnit.MILLISECONDS).build());
+        return new THttpClient(uri.toString(), httpClientBuilder.build());
+    }
+
+    private static Optional<SSLContext> buildSslContext(
```

Comment:
This method is unused.

Reply:
Sorry, this was left over while addressing some of the previous comments. Removed it.